realm/src/androidTest/java/io/realm/RealmInMemoryTest.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm;
+
+import android.test.AndroidTestCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.realm.entities.Dog;
+
+public class RealmInMemoryTest extends AndroidTestCase {
+
+    private final static String IDENTIFIER = "InMemRealTest";
+
+    protected Realm testRealm;
+
+    @Override
+    protected void setUp() throws Exception {
+
+        // Delete the same name realm file just in case
+        RealmConfiguration configuration = new RealmConfiguration.Builder(getContext())
+                .name(IDENTIFIER).build();
+        Realm.deleteRealm(configuration);
+
+        testRealm = Realm.getInMemoryInstance(getContext(), IDENTIFIER);
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        if (testRealm != null) {
+            testRealm.close();
+        }
+    }
+
+    public void testInMemoryRealm() {
+        testRealm.beginTransaction();
+        Dog dog = testRealm.createObject(Dog.class);
+        dog.setName("DinoDog");
+        testRealm.commitTransaction();
+
+        assertEquals(testRealm.allObjects(Dog.class).size(), 1);
+        assertEquals(testRealm.allObjects(Dog.class).first().getName(), "DinoDog");
+        testRealm.close();
+
+        // After all references to the in-mem-realm destroyed,
+        // in-mem-realm with same identifier should create a fresh new instance.
+        testRealm = Realm.getInMemoryInstance(getContext(), IDENTIFIER);
+        assertEquals(testRealm.allObjects(Dog.class).size(), 0);
+    }
+
+    public void testSameIdentifierRealmOnDisk() {
```

Comment:
This is the same test as testDifferentDurabilityThrows, except that here you set a name. For the first test "default.realm" is used as the default name, but thats the only difference as far as I can see. I would move this test to RealmConfigurationTests.